api/submission/fetch_course_report_test.go

```diff
@@ -0,0 +1,110 @@
+package submission
+
+import (
+	"reflect"
+	"testing"
+
+    "github.com/edulinq/autograder/common"
+	"github.com/edulinq/autograder/api/core"
+	"github.com/edulinq/autograder/model"
+	"github.com/edulinq/autograder/report"
+	"github.com/edulinq/autograder/util"
+)
+
+func TestCourseReport(test *testing.T) {
+    testCases := []struct{
+        role model.UserRole
+        courseId string 
```

Comment:
I don't think you even need to set the course.
See the comment below on line 26.